### PR TITLE
SLING-10895 add getPropertyMap() method

### PR DESCRIPTION
**Pull request by joerghoh**

### CODE REVIEW

File: src/main/java/org/apache/sling/api/resource/ResourceResolver.java [hunk 1]
```diff
@@ -842,4 +842,20 @@ Resource copy(final String srcAbsPath,
...
+     *
+     * The resource resolver will clear the map during {@link #close()}, so afterwards the map is empty.
+     * If a stored value implements the {@link #Closeable} interface, the ResourceResolver will invoke the
+     * <code>close()</code> of the value before clearing the map.
+     *
+     * @returns the property map
+     * @since 2.13.0
+     *
+     */
+    public @NotNull Map<String,Object> getPropertyMap();
```
Comment (kwin) [inline src/main/java/org/apache/sling/api/resource/ResourceResolver.java:860]: What about getSessionMap or getCacheMap? That way the intent is clearer from the name.

File: src/main/java/org/apache/sling/api/resource/ResourceResolver.java [hunk 2]
```diff
@@ -842,4 +842,20 @@ Resource copy(final String srcAbsPath,
...
     Resource move(final String srcAbsPath,
               final String destAbsPath) throws PersistenceException;
+
+    /**
+     * Returns a map to store temporary objects.
+     *
+     * This map is suited to store objects which share the same lifecycle as the
+     * resource resolver. The resource resolver itself does not use this map.
+     *
+     * The resource resolver will clear the map during {@link #close()}, so afterwards the map is empty.
```
Comment (kwin) [inline src/main/java/org/apache/sling/api/resource/ResourceResolver.java:852]: Close() may throw exceptions. Are those propagated to the close method of the RR? The javadoc should be clarified in that regard.

File: src/main/java/org/apache/sling/api/resource/ResourceResolver.java (hunk 1 above)
Comment (cziegeler) [reply src/main/java/org/apache/sling/api/resource/ResourceResolver.java:860]: I don't think that the use case is restricted to caching. The best name would probably getAttributesMap but we have read-only attributes already. For me, properties sounds good

File: src/main/java/org/apache/sling/api/resource/ResourceResolver.java (hunk 2 above)
Comment (cziegeler) [reply src/main/java/org/apache/sling/api/resource/ResourceResolver.java:852]: Good point, I would assume they get ignored

Comment (cziegeler) [reply src/main/java/org/apache/sling/api/resource/ResourceResolver.java:858]: updated javadocs

File: src/main/java/org/apache/sling/api/resource/ResourceResolver.java (hunk 1 above)
Comment (cziegeler) [reply src/main/java/org/apache/sling/api/resource/ResourceResolver.java:866]: @kwin  Can we resolve this in one way or the other?

Comment (kwin) [reply src/main/java/org/apache/sling/api/resource/ResourceResolver.java:866]: Ok, let us stick with getPropertyMap()

File: src/main/java/org/apache/sling/api/resource/ResourceResolver.java [hunk 3]
```diff
@@ -607,8 +607,14 @@
      * which was used to create this resolver is no longer active or
      * any of the used resource providers is no longer active.
      *
+     * All objects in the map maintained by {@link #getPropertyMap()} the map implementing
```
Comment (kwin) [inline src/main/java/org/apache/sling/api/resource/ResourceResolver.java:610]:
```suggestion
     * All objects in the map maintained by {@link #getPropertyMap()} implementing
```

File: src/main/java/org/apache/sling/api/resource/ResourceResolver.java [hunk 4]
```diff
@@ -607,8 +607,14 @@
...
+     * the {@link #Closeable} interface, will be closed by calling {@code close()} on each of them.
+     * Any exception throwing by this call will be ignored. The order in which the objects are
```
Comment (kwin) [inline src/main/java/org/apache/sling/api/resource/ResourceResolver.java:612]:
```suggestion
     * Any exception thrown by this call will be ignored. The order in which the objects are
```

**Review by kwin — CHANGES_REQUESTED** [review]

**Review by cziegeler — APPROVED** [review]

File: src/main/java/org/apache/sling/api/resource/ResourceResolver.java [hunk 5]
```diff
@@ -842,4 +848,20 @@ Resource copy(final String srcAbsPath,
...
+     * This map is suited to store objects which share the same lifecycle as the
+     * resource resolver. The resource resolver itself does not use this map.
+     *
+     * The resource resolver will clear the map during {@link #close()}, so afterwards the map is empty.
+     * If a stored value implements the {@link #Closeable} interface, the ResourceResolver will invoke the
+     * <code>close()</code> of the value before clearing the map.
+     *
+     * @returns the property map
+     * @see #close()
+     * @since 2.13.0
```
Comment (kwin) [inline src/main/java/org/apache/sling/api/resource/ResourceResolver.java:864]: Let us add the bundle version as well. Consumers need to know rather that version!

**Review by kwin — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/sling/api/resource/ResourceResolver.java (hunk 5 above)
Comment (kwin) [reply src/main/java/org/apache/sling/api/resource/ResourceResolver.java:864]:
```suggestion
     * @since 2.13.0 (Sling API Bundle 2.24.0)
```

**Review by kwin — APPROVED** [review]

**Comment by joerghoh** [comment]
thanks all for your help!

**Pull request merged**